classes/NewsFetcher.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+use Doctrine\Common\Cache\FilesystemCache;
+use GuzzleHttp\Message\Request;
+use GuzzleHttp\Subscriber\Cache\CacheStorage;
+use GuzzleHttp\Subscriber\Cache\CacheSubscriber;
+use PrestaShop\CircuitBreaker\AdvancedCircuitBreakerFactory;
+use PrestaShop\CircuitBreaker\Contract\FactoryInterface;
+use PrestaShop\CircuitBreaker\FactorySettings;
+use PrestaShop\CircuitBreaker\Storage\DoctrineCache;
+
+class NewsFetcherCore
+{
+    const NUM_ARTICLES = 2;
+
+    const CACHE_DURATION = 86400; // 24 hours
+
+    const CLOSED_ALLOWED_FAILURES = 3;
+    const CLOSED_TIMEOUT_SECONDS = 3;
+
+    const OPEN_ALLOWED_FAILURES = 3;
+    const OPEN_TIMEOUT_SECONDS = 3;
+    const OPEN_THRESHOLD_SECONDS = 86400; // 24 hours
+
+    /** @var array */
+    private $blogSettings;
+
+    /** @var FactoryInterface */
+    private $factory;
+
+    /** @var string */
+    private $isoCode;
+
+    /**
+     * @param string $isoCode
+     */
+    public function __construct(
+        string $isoCode
+    ) {
+        $this->isoCode = $isoCode;
+
+        // Doctrine cache used for Guzzle and CircuitBreaker storage
```

Comment:
ping @jolelievre: I see a command pattern here, maybe can you introduce a "PrestaShopCircuitBreaker" with all that stuff" inside, don't you think ?

Reply:
I'm not sure what you mean, are you refering to a command/handler pattern like in CQRS?